cosmos/models/Workflow.py

```diff
@@ -443,6 +446,19 @@ def delete(self, delete_files=False):
         self.session.commit()
         print >> sys.stderr, '%s Deleted' % self
 
+    def exit_status(self):
+        """
+        Return the exit status of the first failed Task (topologically, not chronologically).
+
+        If no Task failed, return None.
+        """
+        for t in self.task_graph():
+            if t.exit_status:
+                self.log.warning("%s took exit status %s from %s (%s)",
+                                 self, t.exit_status, t, t.status)
+                return t.exit_status
+        return None
+
```

Comment:
I don't think this guarantees the first topological failure (you'd need to toplogical_sort the task_graph).  The answer is also undefined if there are ties - a simple case is if all tasks in a second stage fail, which one is the first failure?  They are topologically identical.  I think you might actually want chronological first failure.. ie sort by finished_on time, and return the first one.

The method could be more general:  Workflow.get_first_failed_task() which should return the Task that failed first, and if no Tasks failed then raise an Exception or return None.  Then the calling code could inspect exit_status.